sbin/docker_cleanup.sh

```diff
@@ -0,0 +1,89 @@
+#!/bin/bash                                                                                                                                               
+
+# An apparent bug in docker may leave stuck zombie networks after
+# autograding has finished.  This docker_cleanup script will be run by
+# the daemon user as root to detect and forceably remove any stuck
+# docker networks.
+
+# this script must be run by root or sudo
+if [[ "$UID" -ne "0" ]] ; then
+    echo "ERROR: This script must be run by root or sudo"
+    exit 1
+fi
+
+mkdir -p "/var/local/submitty/logs/docker_cleanup/"
+logfile_name="/var/local/submitty/logs/docker_cleanup/$(date +%Y%m%d).txt"
+echo  "" >> "${logfile_name}"
+
+# =================================================================================
+function cleanup_dead_networks() {
+    name=${1}
+
+    # if no arguments were passed to this script, check for
+    # autograding on each worker before cleaning up networks
+    if [ $# -eq 0 ]; then
+        # if the autograding_tmp directory is non-empty,
+        # there is active autograding for that worker
+        if test -n "$(ls /var/local/submitty/autograding_tmp/${name})"; then
+            date_time=$(date "+%Y%m%d %H:%M:%S")
+            echo "${date_time} | ${name} has grading in progress" >> "${logfile_name}"
+            return
+        fi
+    fi
+
+    # check for docker networks matching the worker name
+    networks=$(docker network ls | grep "${name}" | awk '{print $2}')
```

Comment:
If docker changes their output format, this may break. Would be more stable to do `docker network ls --format json` and then parse it as JSON.